File: 3_Semestre/PHP-3_Semestre/Lista-03/conta_bancaria.php
<?php

class ContaBancaria {
    public $numeroConta;
    public $titular;
    public $saldo;

    public function __construct($numeroConta, $titular, $saldoInicial) {
        $this->numeroConta = $numeroConta;
        $this->titular = $titular;
        $this->saldo = $saldoInicial;
    }//construtor

    public function depositar($valor) {
        if ($valor > 0) {
            $this->saldo += $valor;
            return "Depósito de R$$valor realizado com sucesso!";
        } else {
            return "Valor inválido para depósito.";
        }
    }//depositar


    public function sacar($valor) {
        if ($valor > 0 && $valor <= $this->saldo) {
            $this->saldo -= $valor;
            return "Saque de R$$valor realizado com sucesso!";
        } else {
        return "Saldo insuficiente ou valor inválido.";
        }
    }//sacar

    public function exibirConta(){
        return "Titular - $this->titular<br>Conta - $this->numeroConta<br>Saldo - R$$this->saldo<br>";
    }//exibirConta

    public function exibirSaldo() {
        return "Saldo atual: R$$this->saldo";
    }
}//class


$conta1 = new ContaBancaria("12345-6", "Bruno", 10000);

echo $conta1->exibirConta() . '<br>';
echo $conta1->exibirSaldo() . "<br>";
echo $conta1->depositar(500) . "<br>";
echo $conta1->exibirSaldo() . "<br>";
echo $conta1->sacar(300) . "<br>";
echo $conta1->exibirSaldo();

?>

File: 3_Semestre/Java-3_Semestre/Visibilidade/src/Encapsulamento/ContaBancaria.java
package Encapsulamento;

import java.util.Scanner;
import java.util.ArrayList;
import java.util.List;

public class ContaBancaria {
	private String titular;
	private double saldo;
	private List<String> historico;
	
	protected ContaBancaria(String titular, double saldo) {
		Scanner input = new Scanner(System.in);
		this.titular = titular;
		this.historico = new ArrayList<String>();
		if (saldo < 0) {
			System.out.println("Você digitou um saldo negativo, deseja alterar o saldo? digite [s/n]: ");
			this.saldo = 0;
			if (input.next().toLowerCase().equals("s")) {
				System.out.print("Digite o novo saldo: ");
				saldo = input.nextDouble();
			}
			this.saldo = saldo < 0 ? 0 : saldo;
		}//if
		else {
			this.saldo = saldo;
		}//else
	}//fim metodo
	public void setTitular(String novoTitular) {
		this.titular = novoTitular;
	}//fim setTitular
	public String getTitular() {
		return this.titular;
	}//fim getTitular
	// Método para obter o saldo
	public double getSaldo() {
	    return this.saldo;
	}
	public void setSaldo(double novoSaldo) {
		this.saldo = novoSaldo;
	}//fim setSaldo
	// Método para obter o histórico (retorna a lista)
	public List<String> getHistorico() {
	    return this.historico;
	}
	public void depositar(double novoDeposito) {
		if (novoDeposito <= 0) {
			System.out.println("Valor Inválido");
		} else {
			this.saldo = this.saldo + novoDeposito;
			historico.add("Deposito no valor de "+ novoDeposito);
			System.out.println("Deposito no valor de " + novoDeposito + "feito com sucesso.");
		}//ifelse
	}//fim depositar
	public void sacar(double valorSaque) {
		if(valorSaque <= 0) {
			System.out.println("O valor digitado não é o esperado!!");
		} else {
			this.saldo = this.saldo - valorSaque;
			historico.add("Saque no valor de "+ valorSaque);
			System.out.println("Saque no valor de "+ valorSaque +"feito com sucesso!");
		}//ifelse
	}//fim sacar
}//fim class
